wasmbinding/message_plugin.go

```diff
@@ -57,11 +57,111 @@ func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddre
 		if contractMsg.RegisterInterchainQuery != nil {
 			return m.registerInterchainQuery(ctx, contractAddr, contractMsg.RegisterInterchainQuery)
 		}
+		if contractMsg.UpdateInterchainQuery != nil {
+			return m.updateInterchainQuery(ctx, contractAddr, contractMsg.UpdateInterchainQuery)
+		}
+		if contractMsg.RemoveInterchainQuery != nil {
+			return m.removeInterchainQuery(ctx, contractAddr, contractMsg.RemoveInterchainQuery)
+		}
 	}
 
 	return m.Wrapped.DispatchMsg(ctx, contractAddr, contractIBCPortID, msg)
 }
 
+func (m *CustomMessenger) updateInterchainQuery(ctx sdk.Context, contractAddr sdk.AccAddress, updateQuery *bindings.UpdateInterchainQuery) ([]sdk.Event, [][]byte, error) {
+	response, err := m.performUpdateInterchainQuery(ctx, contractAddr, updateQuery)
+	if err != nil {
+		ctx.Logger().Debug("performUpdateInterchainQuery: failed to update interchain query",
+			"from_address", contractAddr.String(),
+			"msg", updateQuery,
```

Comment:
just a little question, how does it get printed? marshalled to JSON? I am asking because I don't see `.String()` implementation for `bindings.UpdateInterchainQuery`

Reply:
good question, i've tested, the logger marshals msg to JSON.

Reply:
alright, sounds good enough for me then